x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleIndexerAction.java

```diff
@@ -125,7 +126,19 @@ protected DownsampleIndexerAction.ShardDownsampleRequest newShardRequest(
         ShardRouting shard,
         DownsampleIndexerAction.Request request
     ) {
-        return new DownsampleIndexerAction.ShardDownsampleRequest(shard.shardId(), request);
+        long totalDocCount = client.admin()
```

Comment:
These are blocking calls. I don't think we can do that here.
Maybe we should only focus on the shard doc count?
That we can gather from `RollupShardIndexer`, by invoking `searcher.getIndexReader().numDocs();` in the constructor of that class.